src/daemon/rpmostree-sysroot-upgrader.c

```diff
@@ -1299,6 +1299,21 @@ rpmostree_sysroot_upgrader_deploy (RpmOstreeSysrootUpgrader *self,
 
   if (use_staging)
     {
+      /* touch file *before* we stage to avoid races */
+      if (self->flags & RPMOSTREE_SYSROOT_UPGRADER_FLAGS_LOCK_FINALIZATION)
+        {
+          if (!glnx_shutil_mkdir_p_at (AT_FDCWD,
```

Comment:
I'm not familiar with glnx utils. Is there that a mkdir_p variant which returns the dirfd for the directory it just created, for feeding into the create below?

Reply:
Going to mark this one resolved, but happy to discuss further!

Reply:
Sorry for throwing you into a rabbit hole. On a second thought, I think there are no mkdir variants that directly returns the just-created dirfd, so my question was a silly one.